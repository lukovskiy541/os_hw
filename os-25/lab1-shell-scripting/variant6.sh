#!/bin/bash
d=$1
{ echo ""; find "$d" -mindepth 1 -type d -printf '%P\n'; } |
while read -r r; do
  n=${r:-$(basename "$d")}
  p="$d/$r"
  c=$(find "$p" -type f | wc -l)
  s=$(find "$p" -type f -printf '%s\n' | awk '{t+=$1} END{print t+0}')
  echo -e "$n\t$c\t$s"
done | sort -t $'\t' -k2,2nr -k3,3nr -k1,1 | awk -F'\t' '{print $1" "$2" "$3}'