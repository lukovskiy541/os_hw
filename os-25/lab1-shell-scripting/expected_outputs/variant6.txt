testdata 6 40
beta 2 19
gamma 2 11
gamma/subgamma 2 11
alpha 2 10
